pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -709,68 +709,131 @@ public List<String> getAllResources() {
   }
 
   /**
-   * Get all table names (with type suffix).
+   * Get all table names (with type suffix) in default database.
    *
-   * @return List of table names
+   * @return List of table names in default database
    */
   public List<String> getAllTables() {
+    return getAllTables(null);
+  }
+
+  /**
+   * Get all table names (with type suffix) from provided database.
+   *
+   * @param databaseName database name
+   * @return List of table names in provided database name
+   */
+  public List<String> getAllTables(String databaseName) {
     List<String> tableNames = new ArrayList<>();
     for (String resourceName : getAllResources()) {
-      if (TableNameBuilder.isTableResource(resourceName)) {
+      if (TableNameBuilder.isTableResource(resourceName) && isPartOfDatabase(resourceName, databaseName)) {
         tableNames.add(resourceName);
       }
     }
     return tableNames;
   }
 
+  private boolean isPartOfDatabase(String tableName, String databaseName) {
+    if (databaseName == null) {
+      return StringUtils.split(tableName, '.').length == 1;
+    } else if (databaseName.equalsIgnoreCase(CommonConstants.DEFAULT_DATABASE)) {
+      String[] split = StringUtils.split(tableName, '.');
+      return split.length == 1 || split[0].equalsIgnoreCase(CommonConstants.DEFAULT_DATABASE);
```

Comment:
To simplify the handling, we should prevent table with `default.` prefix. Table should either have non-default database prefix, or no prefix

Reply:
Agree. The APIs will ensure that constraint going forward. Should we drop the ball on existing tables with `default.` prefix?

Reply:
I think that should be okay. I'm not aware of anyone explicitly create table with `default.` prefix, and we should prevent that going forward

Reply:
Thanks for confirming. Removed the check